content/blog/eng/on-living-documentation.md

```diff
@@ -0,0 +1,194 @@
+---
+Today I want to talk about living documentation, having just finished the aptly-named book by Cyrille Martraire, **[Living Documentation: continuous knowledge sharing by design](https://www.pearson.com/store/p/living-documentation-continuous-knowledge-sharing-by-design/P100000797747/9780134689326)**, published by Pearson.
+
+# The need for documentation
+Documentation supplements the knowledge we might not have.
+Lack of knowledge manifests in
+- wasted time (finding the missing points or guessing them)
+- biased decisions due to this lack
+
+- Hint: when you don’t know something, you usually don’t know of not knowing ;)
+
+So the time spent on harvesting knowledge should be considered helping build the stakeholders’ application mental model. And this is important because is that model that developers will use to augment the code, product owners will use to describe the stories to implements, and business owners will use to describe their objectives and key results….
+
+Let’s hope that every stakeholder has a shareable and overlapping mental model!
+
+#A little confession and the need for something better
+
+I confess: I like to write documentation.
+
+I admit: I am (sometimes) dedicated to the antipattern: human dedication (see page 32).
+
+Sometimes I need to sketch up things while following the execution of the code, and it’s a pity to throw away those findings.
+
+Other times I need to clear up my mind over some convoluted code.
+
+In any case, I resort to the tool that the company I work for offers: Confluence pages.
+
+Once I’ve done writing the article, that article begins to age, eventually reaching, as the author says, the information graveyard.
+
+If I’m lucky enough, I can intercept changes to the code and amend the docs accordingly.
+
+But not everyone is so fond of writing documentation.
+
+And that’s where this book comes in handy: ideas on automating the generation of documentation.
+
+#What this book is about and what it is not
+
+This book is not about the end-user documentation, nor does aim to bookish-style, well-refined output.
+
+This book is about sharing of knowledge and making it available in the right place, right on time.
+
+By doing this, it hopes to shed a light on the overall design of the application being documented, eventually triggering an opportunity for refactoring.
+
+Overall, it wishes the code to be so clear and self-documented that the additional documentation became useless and redundant.
+
+And that’s why this is not a book about end-user documentation: not every audience is able/willing to read the code, nor that every audience should do it.
+
+#Written documentation is not always the correct answer
+
+The book follows the track of the agile / TDD / DDD / BTD development and builds on their pillars.
+
+> With TDD, the tests are first considered as specifications. With DDD, we identify the code and the model of the business domain […]. … we expect the code to tell the whole story about the domain.
```

Comment:
with TDD and BDD

Reply:
DDD che cosa era nella mia testa? un Developer-Driven-Development ?? LOL